### DISCUSSION TITLE
What is your favourite light theme?

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#1118] [General]
Hi. 
Since the themes are not sorted by usage/favourite, what is your favourite light theme?

**Comment by wez** [comment]
I think most users prefer dark themes.  I believe that @EpocSquadron, @jankatins and @fanzeyi use dark/light theme auto switching and may have some suggestions!

**Reply by EpocSquadron** [reply]
I actually am pretty new to light themes myself, but I combed through most of the available ones today and find that CLRS is my favorite. Material is also pretty nice. Seems harder to construct a good light theme than a good dark theme.

**Reply by azzamsa** [reply]
Thanks a lot for the input. 

> Seems harder to construct a good light theme than a good dark theme.

Couldn't agree more.

**Reply by fanzeyi** [reply]
I use `tomorrow` from `base16` for both light and dark theme. `base16-tomorrow` and `base16-tomorrow-night-eighties`

**Reply by azzamsa** [reply]
Thanks for the addition @fanzeyi

**Reply by xpufx** [reply]
> use dark/light theme auto switching

How is this done?

**Reply by wez** [reply]
https://wezfurlong.org/wezterm/config/lua/window/get_appearance.html shows how to switch theme based on the OS light/dark appearance

**Reply by xpufx** [reply]
This is great thank you. I only use light mode in my OS (windows) though. Could I bind a key combination to switch the color scheme? I didn't find a suitable action on https://wezfurlong.org/wezterm/config/lua/keyassignment/index.html . 

EDIT: I just saw and am checking out [Dynamic Color Escape Sequences](https://wezfurlong.org/wezterm/config/appearance.html#dynamic-color-escape-sequences)

**Reply by xpufx** [reply]
This is how I was able to do it.

```lua
-- toggle light/dark scheme with CTRL+l
wezterm.on("toggle-dark-mode", function(window,pane)
  local light_scheme = "Builtin Solarized Light"
  local dark_scheme = "Builtin Solarized Dark"
  local overrides = window:get_config_overrides() or {}
  wezterm.log_info("Current color scheme is: ", overrides.color_scheme)
  if (overrides.color_scheme == light_scheme)
  then
  wezterm.log_info("Setting to Dark Scheme: ", overrides.color_scheme)
  overrides.color_scheme = dark_scheme

  else
  wezterm.log_info("Setting to Light ", overrides.color_scheme)
  overrides.color_scheme = light_scheme
  
  end
  window:set_config_overrides(overrides)
end)

return {
  keys = {
    {
	key="l", mods="CTRL", action=wezterm.action{EmitEvent="toggle-dark-mode"},
  },
}
```

**Reply by xpufx** [reply]
Thanks for the script coloring fix. I added this to Show&Tell with a short blurb.
